### PR TITLE
Fix review suggestions for #29814

### PR DESCRIPTION
**Pull request by lasinicl**
## Purpose
$title

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/30128

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/types/never/never_type_runtime.bal [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+|};
+
+function testNeverRuntime11() {
+    Record x = { i: 1, j: [] };
+    boolean b = x is never;
+    assertEquality(false, b);
+}
+
+function testNeverRuntime12() {
+    Record x = { i: 1, j: [] };
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/never/never_type_runtime.bal:98]:
```suggestion
    Record x = {i: 1, j: []};
```

Please check other similar use in these changes too.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java [hunk 2]
```diff
@@ -557,7 +557,9 @@ public static boolean checkIsType(Type sourceType, Type targetType, List<TypePai
             return true;
         }
 
-        if (checkIsNeverTypeOrStructureTypeWithARequiredNeverMember(sourceType)) {
+        Set<Type> visitedTypeSet = new HashSet<>();
+        visitedTypeSet.add(sourceType);
+        if (checkIsNeverTypeOrStructureTypeWithARequiredNeverMember(sourceType, visitedTypeSet)) {
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java:562]: Please check the same for the compile-time check also.

**Review by MaryamZi — APPROVED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaryamZi) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java]: @lasinicl, we are still doing this check in checkIsType right? Can you please create an issue for this?

[the remark this answers is not in the document]
Comment (lasinicl) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java]: Created https://github.com/ballerina-platform/ballerina-lang/issues/30315

**Pull request merged**